From 663a14423baea0e05ba79d90d2497dde5e4594bd Mon Sep 17 00:00:00 2001
From: "Gary S. Robertson" <gary.robertson@linaro.org>
Date: Thu, 11 Sep 2014 13:02:47 -0500
Subject: [LTP][PATCH] Realtime tests: Fix robust mutex conditionals

sbrk_mutex, testpi-5, and testpi-6 realtime tests in subdir
testcases/realtime/func/pi-tests used compile time config variables
which were not generated by autoconf in order to configure tests
for robust mutexes.  Changed these conditionals to use the config
variables actually generated in the autoconf process.

Upstream-Status: Submitted

Signed-off-by: Gary S. Robertson <gary.robertson@linaro.org>
---
 testcases/realtime/func/pi-tests/sbrk_mutex.c |    2 +-
 testcases/realtime/func/pi-tests/testpi-5.c   |    2 +-
 testcases/realtime/func/pi-tests/testpi-6.c   |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/testcases/realtime/func/pi-tests/sbrk_mutex.c b/testcases/realtime/func/pi-tests/sbrk_mutex.c
index 684021f..5c325b4 100644
--- a/testcases/realtime/func/pi-tests/sbrk_mutex.c
+++ b/testcases/realtime/func/pi-tests/sbrk_mutex.c
@@ -45,7 +45,7 @@
 #include <unistd.h>
 #include "librttest.h"
 
-#if defined(HAS_PTHREAD_MUTEXTATTR_ROBUST_APIS) && defined(PTHREAD_MUTEX_ROBUST_NP)
+#if HAS_PTHREAD_MUTEXTATTR_ROBUST_APIS
 
 #define NUM_MUTEXES 5000
 #define NUM_THREADS 50
diff --git a/testcases/realtime/func/pi-tests/testpi-5.c b/testcases/realtime/func/pi-tests/testpi-5.c
index a1d93cc..70f02fd 100644
--- a/testcases/realtime/func/pi-tests/testpi-5.c
+++ b/testcases/realtime/func/pi-tests/testpi-5.c
@@ -69,7 +69,7 @@ int do_test(int argc, char **argv)
 	pthread_mutexattr_t mutexattr;
 	int retc, protocol;
 
-#if HAS_PTHREAD_MUTEXATTR_PROTOCOL_FUNCTIONS
+#if HAS_PTHREAD_MUTEXTATTR_ROBUST_APIS
 
 	if (pthread_mutexattr_init(&mutexattr) != 0)
 		printf("Failed to init mutexattr\n");
diff --git a/testcases/realtime/func/pi-tests/testpi-6.c b/testcases/realtime/func/pi-tests/testpi-6.c
index b3c3e4a..f715eee 100644
--- a/testcases/realtime/func/pi-tests/testpi-6.c
+++ b/testcases/realtime/func/pi-tests/testpi-6.c
@@ -41,7 +41,7 @@
 #include <unistd.h>
 #include <librttest.h>
 
-#if defined(PTHREAD_MUTEX_ROBUST_NP)
+#if HAS_PTHREAD_MUTEXTATTR_ROBUST_APIS
 pthread_mutex_t child_mutex;
 
 void *child_thread(void *arg)
-- 
1.7.9.5

